### PR TITLE
Improve CPU use of parallel test runs

### PR DESCRIPTION
**Pull request by pitrou**
This shrinks the test suite runtime from 3:16 to 1:47 here.

### CODE REVIEW

File: numba/tests/__init__.py [hunk 1]
```diff
@@ -376,39 +376,25 @@ def _run_inner(self, result):
...
-                try:
-                    child_result = ticket.get(120)
-                except TimeoutError:
-                    print("TimedOut", test)
-                    raise
-                result.add_results(child_result)
-                if child_result.shouldStop:
-                    return
-
+                child_result = it.__next__(120)
```
Comment (sklam) [inline numba/tests/__init__.py]: Did I miss that API or is it undocumented?

**Comment by sklam** [comment]
Yes, the Travis 2.6 build finished in 3 minutes instead of 8 minutes.

File: numba/tests/__init__.py (hunk 1 above)
Comment (pitrou) [inline numba/tests/__init__.py]: It's documented but not very visible:
https://docs.python.org/2.6/library/multiprocessing.html#multiprocessing.pool.multiprocessing.Pool.imap

Comment (sklam) [inline numba/tests/__init__.py]: Yes, I see it now. Thanks!

**Pull request merged**